### DISCUSSION TITLE
Stream/Queue size in bytes (x-max-length-bytes)

### DISCUSSION THREAD

**Discussion by kurenchuksergey** [spring-projects/spring-amqp#2551] [General]
Hi, 
I have checked rabbitmq documentation again and noticed that x-max-length-bytes are limited by int value only for the queues (https://www.rabbitmq.com/maxlength.html).

But for the streams limitation is not described and even thought in examples (https://www.rabbitmq.com/streams.html#retention)
there is a value greater than the integer (20 GB).
Unfortunately, the max size isn't described (or I didn't find it). Can I try to add an additional overloaded method to SuperStreamBuilder? 

`
public SuperStreamBuilder maxLength(long bytes) {
		return withArgument("max-length-bytes", bytes);
}
`

And maybe the same for x-stream-max-segment-size-bytes? Its limit is also unclear
Thanks

**Comment by garyrussell** [comment]
I discussed it with the RabbitMQ team and an Erlang `int` is 8 bytes, so should be `long` in Java.

https://github.com/spring-projects/spring-amqp/issues/2552
